source/speech/manager.py

```diff
@@ -290,43 +291,55 @@ def _queueSpeechSequence(self, inSeq: SpeechSequence, priority: Spri) -> bool:
+			paramsToReplay.extend(paramTracker.getChanged())
+		else:
+			lastOutSeq = outSeqs[-1] if outSeqs else None
+		lastCommand = lastOutSeq[-1] if lastOutSeq else None
+		if lastCommand is None or isinstance(lastCommand, (EndUtteranceCommand, ConfigProfileTriggerCommand)):
+			# It doesn't make sense to start with or repeat EndUtteranceCommands.
+			# We also don't want an EndUtteranceCommand immediately after a ConfigProfileTriggerCommand.
+			return
+		if not isinstance(lastCommand, IndexCommand):
+			# Add an index so we know when we've reached the end of this utterance.
+			reachedIndex = next(self._indexCounter)
+			lastOutSeq.append(IndexCommand(reachedIndex))
+		outSeqs.append([EndUtteranceCommand()])
+
 	def _processSpeechSequence(self, inSeq: SpeechSequence):
 		paramTracker = ParamChangeTracker()
 		enteredTriggers = []
 		outSeqs = []
 		paramsToReplay = []
-
-		def ensureEndUtterance(seq: SpeechSequence):
-			# We split at EndUtteranceCommands so the ends of utterances are easily found.
-			# This function ensures the given sequence ends with an EndUtterance command,
-			# Ensures that the sequence also includes an index command at the end,
-			# It places the complete sequence in outSeqs,
-			# It clears the given sequence list ready to build a new one,
-			# And clears the paramsToReplay list
-			# and refills it with any params that need to be repeated if a new sequence is going to be built.
-			if seq:
-				# There have been commands since the last split.
-				lastOutSeq = paramsToReplay + seq
-				outSeqs.append(lastOutSeq)
-				paramsToReplay.clear()
-				seq.clear()
-				# Re-apply parameters that have been changed from their defaults.
-				paramsToReplay.extend(paramTracker.getChanged())
-			else:
-				lastOutSeq = outSeqs[-1] if outSeqs else None
-			lastCommand = lastOutSeq[-1] if lastOutSeq else None
-			if lastCommand is None or isinstance(lastCommand, (EndUtteranceCommand, ConfigProfileTriggerCommand)):
-				# It doesn't make sense to start with or repeat EndUtteranceCommands.
-				# We also don't want an EndUtteranceCommand immediately after a ConfigProfileTriggerCommand.
-				return
-			if not isinstance(lastCommand, IndexCommand):
-				# Add an index so we know when we've reached the end of this utterance.
-				reachedIndex = next(self._indexCounter)
-				lastOutSeq.append(IndexCommand(reachedIndex))
-			outSeqs.append([EndUtteranceCommand()])
+		currentSynth = getSynth()
 
 		outSeq = []
 		for command in inSeq:
+			if isinstance(command, LangChangeCommand) and currentSynth.name == 'oneCore':
+				langCode = command.lang.split('_')[0]
+				langSupported = False
+				currentSynthLanguages = currentSynth.availableLanguages
+				for lang in currentSynthLanguages:
+					if lang and normalizeLanguage(lang).split('_')[0] == langCode:
+						langSupported = True
+				if not langSupported:
+					log.warning(f"Language {command.lang} not supported by {currentSynth.name} ({currentSynthLanguages})")
```

Comment:
There are tests for this code `tests/unit/test_speechManager/__init__.py` unfortunately they are complicated. Please add new tests for this changed behavior.

Reply:
The only intended change of behaviour is introduced logging. Is there a standard way to test this in NVDA?

Reply:
If the changes to SpeechManager are only to extend logging, then don't worry about unit tests. I only looked at this quickly and assumed it was more extensive.